### PR TITLE
Move turbconv fields from aux to cache

### PR DESCRIPTION
**Pull request by charleskawczynski**
### Description

This PR removes some turbconv aux fields, and instead computes them in (arbitrarily) the first order flux, so that it will be available in `single_stack_diagnostics`.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: test/Atmos/EDMF/edmf_kernels.jl [hunk 1]
```diff
@@ -882,12 +841,26 @@ function precompute(::EDMF, bl, args, ts, ::Flux{FirstOrder})
...
+    a_up = vuntuple(N_up) do i
+        fix_void_up(up[i].ρa, up[i].ρa * ρ_inv)
+    end
+    if !(bl.moisture isa DryModel)
+        q_tot_up = vuntuple(N_up) do i
+            fix_void_up(up[i].ρa, up[i].ρaq_tot / up[i].ρa, gm.moisture.ρq_tot)
+        end
+    end
+
+    return (; env, a_up, q_tot_up, ρa_up, θ_liq_up, w_up, fix_void_up)
```
Comment (ilopezgp) [inline test/Atmos/EDMF/edmf_kernels.jl:863]: I think `fix_void_up` is not needed in the return, I probably left this here in some PR.

**Comment by ilopezgp** [comment]
@charleskawczynski do you think we can fix the exportability issue with the `single_stack_diagnostics` in this PR?

**Comment by charleskawczynski** [comment]
No, it’s pretty separate work. I started working on it, though

File: test/Atmos/EDMF/edmf_kernels.jl [hunk 2]
```diff
@@ -733,10 +643,9 @@ function source(::GradProdSource{en_ρaθ_liq_q_tot_cv}, atmos, args)
 end
 
 function source(::BuoySource{up_ρaw{i}}, atmos, args) where {i}
-    # TODO: Cache buoyancy
+    @unpack buoy = args.precomputed.turbconv
     up = args.state.turbconv.updraft
-    up_aux = args.aux.turbconv.updraft
-    return up[i].ρa * up_aux[i].buoyancy
+    return up[i].ρa * buoy.up[i]
```
Comment (ilopezgp) [inline test/Atmos/EDMF/edmf_kernels.jl:648]: Should this tendency also be wrapped in `fix_void_up`>

**Review by ilopezgp — APPROVED** [review]

**Comment by ilopezgp** [comment]
LGTM

File: test/Atmos/EDMF/edmf_kernels.jl (hunk 2 above)
Comment (charleskawczynski) [reply test/Atmos/EDMF/edmf_kernels.jl:648]: I’ll let @yairchn address this in #1927

Comment (yairchn) [reply test/Atmos/EDMF/edmf_kernels.jl:648]: #1927 is still a draft PR - should I make it a PR ?

Comment (yairchn) [reply test/Atmos/EDMF/edmf_kernels.jl:648]: also buoy.up[I] is computed from ts.up which is protected with up[i].ρa

Comment (charleskawczynski) [reply test/Atmos/EDMF/edmf_kernels.jl:648]: We’ll probably need to rebase #1927 once this merges

**Pull request merged**